src/Controls/src/Core/Platform/Android/Extensions/ButtonExtensions.cs

```diff
@@ -41,19 +41,30 @@ public static void UpdateContentLayout(this MaterialButton materialButton, Butto
 				// ourselves for our custom implemented IconGravityBottom
 				materialButton.IconPadding = (int)context.ToPixels(contentLayout.Spacing);
 
+				// For IconGravityTextEnd and IconGravityTextStart, setting the Icon twice
```

Comment:
Because we aren't always clearing and resetting the Icon I needed to add this code back in order for the Icon code to work correctly again.

Reply:
To make things better, maybe we can just put this code inside the IconGravity override: https://github.com/dotnet/maui/blob/05078ba9836c48266fefd674c18c566248c2b7fa/src/Core/src/Platform/Android/MauiMaterialButton.cs#L28-L37

This will keem the calling code much simpler.